### PR TITLE
Create edit_reports permission for configurable reports

### PR DESCRIPTION
**Pull request by solleks**
I'm not sure this change is ready for review, but I want to get a sanity check and see if some tests will run on the PR.

[Feature request document.](https://docs.google.com/document/d/1bmo7swOiXyUTW6IcWn9HFCNcdUS8m6KzMcSK2biN7zg/edit?ts=5ef3962a)

[Trello card.](https://trello.com/c/YjJacM4G/83-detach-reports-from-data-permissions)

##### SUMMARY
This change creates a new permission ‘edit_reports’ that controls two actions - creating custom reports and editing custom reports. The pieces of the change are:

- Add the ‘edit_reports’ property to the corehq.apps.users.models.Permissions class.
- Add a migration for the new property that copies the ‘edit_data’ property.
- Change the following permissions checks from ‘edit_data’ to ‘edit_reports’
- Whether the “Report Builder” section appears in the ProjectReportsTab
- The method_decorator on ReportBuilderView.dispatch
- Propagate ‘edit_reports’ to page_context of ConfigurableReportView to avoid showing the ‘Edit Report’ button.
- Add ‘Reports’ section to edit role modal dialog, which contains Create and Edit Reports checkbox for ‘edit_reports’ permission and existing Access All Reports and Access Specific Reports controls.
EditRole
- Change Reports column of project ‘settings/users/roles/’ page to show a check mark when role has ‘edit_reports’ permission and a view only icon when role has only ‘view_reports’ permission.
RolePermissions

##### FEATURE FLAG
Not specific to a feature flag.

##### RISK ASSESSMENT / QA PLAN
This change should get a QA plan as it will affect users permissions to take actions.
I need to learn how to set up a QA plan.

##### PRODUCT DESCRIPTION
This change allows project administrators to create roles that can edit data but not create/edit reports and vice versa. They will make the choice in the edit role dialog and see the effect in the role and permissions page.

### CODE REVIEW

File: corehq/apps/users/migrations/0019_editreportspermissions.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    ).all()
+    for role_doc in iter_docs(UserRole.get_db(), [r['id'] for r in roles]):
+        role = UserRole.wrap(role_doc)
+
+        changed = False
+        if role.permissions.edit_data:
+            role.permissions.edit_reports = True
+            changed = True
+        elif role.permissions.edit_reports:
+            role.permissions.edit_reports = False
```
Comment (proteusvacuum) [inline corehq/apps/users/migrations/0019_editreportspermissions.py:24]: I'm not sure I follow why this is being set to `False` here.

**Review by proteusvacuum — COMMENTED** [review]
Hey! First HQ PR :tada: !

This looks really great. Sanity, checked! 

Did you mean to commit the submodule changes? Sometimes these diverge from the branch you are working on, and it doesn't seem like the changes in those are related to this PR. 

To make a QA request you can fill out this form: https://docs.google.com/forms/d/e/1FAIpQLSehQl5VXuHED_HLXaMcUoWleBGqpZu_chPXfWaxdcb0QbDh4Q/formResponse and the QA team will reach out to you if they need clarifications.

We normally deploy QA branches to staging - instructions for how to do that are found in `scripts/staging.yaml`, but feel free to reach out in #staging on slack if you have questions about that, I'm not sure if you have deploy permissions on the staging machines. 

Feel free to reach out to me on slack if you wanted to chat through anything else, although this seems pretty complete already.

**Comment by solleks** [comment]
I think I just wrecked this PR by not committing a set of changed files after a merge. I will start over in a new branch.

File: corehq/apps/users/migrations/0019_editreportspermissions.py (hunk 1 above)
Comment (solleks) [reply corehq/apps/users/migrations/0019_editreportspermissions.py:24]: I borrowed this logic from the web apps permission change, which was added after the initial change:
https://github.com/dimagi/commcare-hq/blame/master/corehq/apps/users/migrations/0016_webappspermissions.py

It makes sure that edit_reports won't be true if edit_data is false.

**Pull request closed without merging**